src/Sylius/Component/Core/Resolver/DefaultShippingMethodResolver.php

```diff
@@ -33,27 +35,20 @@ class DefaultShippingMethodResolver implements DefaultShippingMethodResolverInte
     private $shippingMethodRepository;
 
     /**
-     * @var ZoneMatcherInterface|null
+     * @var ZoneMatcherInterface
      */
     private $zoneMatcher;
 
     /**
      * @param ShippingMethodRepositoryInterface $shippingMethodRepository
-     * @param ZoneMatcherInterface|null $zoneMatcher
+     * @param ZoneMatcherInterface $zoneMatcher
      */
     public function __construct(
         ShippingMethodRepositoryInterface $shippingMethodRepository,
-        ?ZoneMatcherInterface $zoneMatcher = null
+        ZoneMatcherInterface $zoneMatcher
```

Comment:
We should remove this argument (revert #8629 changes), the current version is a BC break :)